proxy/src/pancake_proxy.cpp

```diff
@@ -8,7 +8,7 @@ void pancake_proxy::init(const std::vector<std::string> &keys, const std::vector
     real_distribution_ = *(distribution *)args[0];
```

Comment:
such pointer conversion is really bad practice, and is just asking for trouble; look at static_cast, dynamic_cast and reinterpret_cast